.github/workflows/lint-and-test.yml

```
@@ -0,0 +1,40 @@
+name: Run Tests and Linter
+
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+
+jobs:
+  lint-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.8.18, 3.9.18, 3.10.13, 3.11.8]
+        django-version: [3.2, 4.2]
+
+    name: Lint and Test (Python ${{ matrix.python-version }} - Django ${{ matrix.django-version }})
+
+    steps:
+      - uses: actions/checkout@v2
```

```suggestion
      - uses: actions/checkout@v4
```
this should squelch the deprecation notices